contract/edenexplorer/src/edenexplorer.cpp

```diff
@@ -0,0 +1,56 @@
+#include <eosio/asset.hpp>
+#include <eosio/eosio.hpp>
+
+#include <edenexplorer.hpp>
+
+namespace eden {
+
+  void edenexplorer_contract::categorize( eosio::name  account,
+                                          std::string &new_memo,
+                                          std::string &tx_id ) {
+    require_auth( account );
+
+    eosio::action{ { get_self(), "active"_n },
+                   "isedenmember"_n,
+                   "checkmember"_n,
+                   std::tuple{ account } }
+        .send();
+
+    eosio::check( parse_memo( new_memo ),
+                  "Incorrect format or category, use: 'eden_expense: "
+                  "<category>/<description>'" );
+    eosio::check( tx_id.length() == 64, "Incorrect length transaction id" );
+  }
+
+  bool edenexplorer_contract::exist_category( std::string &category_name ) {
+    int         pos_space = category_name.find( " " );
+    std::string category =
+        pos_space == -1
+            ? category_name
+            : category_name.substr( pos_space + 1, category_name.length() );
+
+    return ( std::find( categories.begin(), categories.end(), category ) !=
```

Comment:
Let's optimize this logic.

Let's use [any_of](https://cplusplus.com/reference/algorithm/any_of/) instead of `find`. `find` returns an iterator of the value in the array, we don't need that, instead, we only need to know if it contains the value, if so, get the true value right away.